python/lsst/cbp/coordinateConverter.py

```diff
@@ -0,0 +1,856 @@
+            observed=telRotObserved,
+            offset=self.config.telRotOffset,
+            scale=self.config.telRotScale)
+
+    @property
+    def cbpAzAltInternal(self):
+        """Internal az/alt of the CBP (read only)
+
+        Data type is `lsst.geom.SpherePoint`
+
+        Primarily intended for testing
+        """
+        return self._cbpAzAlt
+
+    @property
+    def telAzAltInternal(self):
+        """Internal az/alt of the telescope (read only)
+
+        Data type is `lsst.geom.SpherePoint`
+
+        Primarily intended for testing
+        """
+        return self._telAzAlt
+
+    @property
+    def telRotInternal(self):
+        """Internal angle of the telescope camera rotator (read only)
+
+        Data type is `lsst.geom.Angle`
+
+        Primarily intended for testing
+        """
+        return self._telRot
+
+    def setPupilFieldAngle(self, pupilPos, pupilFieldAngle=None, beam=None):
+        """Set the pupil field angle and pupil position of a beam
+
+        Compute new telescope, camera rotator and CBP positions
+        and thus update beam info.
+
+        This method is primarily intended for internal use,
+        to support the other set methods. It is public so it can be
+        unit-tested.
+
+        Parameters
+        ----------
+        pupilPos : pair of `float`
+            Position of the specified beam on the
+            :ref:`telescope pupil <lsst.cbp.pupil_position>` (x, y mm)
+        pupilFieldAngle : pair of `float` (optional)
+            Pupil field angle of specified beam (x, y rad);
+            defaults to (0, 0)
+        beam : `int` or `str` (optional)
+            Name or index of beam; defaults to self.maskInfo.defaultBeam
+        """
+        beam = self.maskInfo.asHoleName(beam)
+        if pupilFieldAngle is None:
+            pupilFieldAngle = Point2D(0, 0)
+        else:
+            pupilFieldAngle = Point2D(*pupilFieldAngle)
+        beamPosAtCtr = coordUtils.computeShiftedPlanePos(pupilPos, pupilFieldAngle,
+                                                         -self.config.telPupilOffset)
+        beamVectorInCtrPupil = self._computeBeamVectorInCtrPupilFrame(
+            beamPosAtCtr=beamPosAtCtr, pupilFieldAngle=pupilFieldAngle)
+        cbpVectorInCtrPupil = self._computeCbpVectorInCtrPupilFrame(
+            beamPosAtCtr=beamPosAtCtr,
+            beamVectorInCtrPupil=beamVectorInCtrPupil)
+
+        telAzAlt = coordUtils.computeAzAltFromBasePupil(
+            vectorBase=self.config.cbpPosition,
+            vectorPupil=cbpVectorInCtrPupil)
+
+        beamVectorBase = coordUtils.convertVectorFromPupilToBase(
+            vectorPupil=beamVectorInCtrPupil,
+            pupilAzAlt=telAzAlt,
+        )
+        beamFieldAngleCbp = self._getBeamCbpFieldAngle(beam)
+        beamUnitVectorCbpPupil = coordUtils.fieldAngleToVector(beamFieldAngleCbp, self.config.cbpFlipX)
+        cbpAzAlt = coordUtils.computeAzAltFromBasePupil(
+            vectorBase=-beamVectorBase,
```

Comment:
I tried this out but I'm not happy with the results because it is hard to apply consistently. For instance sometimes set to negative scalars and those clearly can't fit this pattern.

I wish we could have spaces around the `=` as that would make things much clearer.

Reply:
Yes, spaces would indeed have been my first choice too. Not to worry then 🙂